src/application/pokegear/radio/shows/commercials.c

```diff
@@ -0,0 +1,157 @@
+#include "global.h"
+
+#include "application/pokegear/radio/radio_internal.h"
+#include "msgdata/msg.naix"
+#include "msgdata/msg/msg_0412.h"
+
+#include "math_util.h"
+#include "sound_02004A44.h"
+#include "sound_radio.h"
+#include "sys_flags.h"
+#include "unk_02005D10.h"
+
+typedef enum CommercialUnlockFlag {
+    COMM_UNLOCK_ALWAYS,
+    COMM_UNLOCK_POKEDEX,  // effectively always
+    COMM_UNLOCK_ECRUTEAK, // most of these are fly unlocks
+    COMM_UNLOCK_CIANWOOD,
+    COMM_UNLOCK_OLIVINE,
+    COMM_UNLOCK_GAME_CLEAR,
+    COMM_UNLOCK_VERMILION,
+    COMM_UNLOCK_RESTORED_POWER,
+} CommercialUnlockFlag;
+
+typedef struct CommercialsData {
+    HeapID heapID;
+    u16 state;
+    u16 msgID;
+    u8 commercialUnlockFlags[8];
+    u8 unlockedCommercials[36];
+} CommercialsData;
+
+void RadioShow_Commercials_Init(RadioShow *radioShow);
+void RadioShow_Commercials_Unload(RadioShow *radioShow);
+
+// Byte 0: Event condition (see CommercialUnlockFlag)
+// Byte 1: Whether the commercial plays in Johto (1), Kanto (2), or both (3)
+// Byte 2: Which channel the commercial will play on (0xFF = any)
+static const u8 sCommercialsData[][3] = {
+    { COMM_UNLOCK_ALWAYS,         3, 0xFF },
```

Comment:
an idea: (you don't need to address this)
```suggestion
    { COMM_UNLOCK_ALWAYS,         (JOHTO | KANTO), 0xFF },
```